### PR TITLE
MPP-2721: Handle uppercase letters in domain adresses

### PR DESCRIPTION
**Pull request by jwhitlock**
When a domain address is created on the website, it is always lowercase. When a domain address is created in the field (such as asking for an email to send a receipt for checkout), it may include capital letters, as noted in MPP-2721. This PR allows capital letters in the local portion of a domain address, so that these are equivalent:

* `hotel@mysubdomain.mozmail.com` (how it appears in the database and the dashboard)
* `Hotel@mysubdomain.mozmail.com`
* `HOTEL@mysubdomain.mozmail.com`

Capitals in the subdomain portion are not allowed by the code, but are also not allowed domain names, so (most?) mail clients and server would also allow:

* `Hotel@MySubdomain.mozmail.com`

In addition:

* Wrap some long lines to fix flake8
* Re-write the `GetAddressTest` tests, to fix some existing issues (such as tests passing when expected exception is not raised) and some aesthetic issues (test order, longer than needed test names, consistent names)

## How to test

* Get emails working locally, with a custom subdomain and a domain address, and with `./manage.py process_emails_from_sqs` running
* Send an email to an existing domain address, but with a capital letter in the local part (for example, `Domain@subdomain.mymail.com` for `domain@subdomain.mymail.com`)
    * [x] The email is delivered to the true address
    * [x] The dashboard Forwarded count is incremented (after refresh)
* Send an email to a new domain address, with a capital letter in the local part (for example, `NewAddress@subdomain.mymail.com`)
    * [x] The email is delivered to the true address
    * [x] The dashboard shows the new domain address with lower-case letters (after refresh)
* Send an email to with a capital letter in the subdomain part (for example, `TryThis@Subdomain.mymail.com`)
    * [x] The email is delivered to the true address
    * [x] The dashboard shows the new domain address with lower-case letters (after refresh)

## Checklist

- [x] I've added unit tests to test for potential regressions of this bug.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

File: emails/views.py [hunk 1]
```diff
@@ -846,6 +848,7 @@ def _get_domain_address(local_portion, domain_portion):
     if address_domain != get_domains_from_settings()["MOZMAIL_DOMAIN"]:
         incr_if_enabled("email_for_not_supported_domain", 1)
         raise ObjectDoesNotExist("Address does not exist")
+    local_address = local_portion.lower()
```
Comment (groovecoder) [inline emails/views.py:851]: praise: 🎉

**Review by groovecoder — APPROVED** [review]

**Comment by jwhitlock** [comment]
New changes:

* Expanded lowercasing to relay masks, domains in general, and the replies email
* Changed the interface to `_get_address` - it now takes the email address and does its own splitting.

**Review by groovecoder — APPROVED** [review]

**Comment by jungle-boogie** [comment]
Will this handle the case where email addresses may be typed with uppercase letters and not be received?

Relay@mozmail.com will not be received but relay@mozmail.com will.

**Comment by Vinnl** [comment]
@jungle-boogie Yep, although it'd be `Relay@yoursubdomain.mozmail.com` - the `@mozmail.com` address don't allow coming up with your own addresses.

(Also note that this isn't live yet; if everything goes smoothly, the fix should be published February 7.)

**Pull request merged**